### ISSUE TITLE
loudness-scanner fails to compile in fedora-28 beta

### ISSUE THREAD

**Issue by zuzzurro** [jiixyj/loudness-scanner#29]
```make
2184> make -k
[  5%] Built target ebur128
[ 11%] Built target ebur128_static
[ 22%] Built target scanner-common
[ 30%] Built target scanner-tag
[ 38%] Built target scanner-lib
[ 44%] Built target filetree
[ 50%] Built target input
[ 55%] Built target loudness
[ 63%] Built target input_sndfile
[ 69%] Built target input_mpg123
[ 72%] Building C object scanner/inputaudio/ffmpeg/CMakeFiles/input_ffmpeg.dir/input_ffmpeg.c.o
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:16:31: error: ‘FF_INPUT_BUFFER_PADDING_SIZE’ undeclared here (not in a function); did you mean ‘AV_INPUT_BUFFER_PADDING_SIZE’?
 #define BUFFER_SIZE (192000 + FF_INPUT_BUFFER_PADDING_SIZE)
                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:31:16: note: in expansion of macro ‘BUFFER_SIZE’
   float buffer[BUFFER_SIZE / 2 + 1];
                ^~~~~~~~~~~
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c: In function ‘ffmpeg_open_file’:
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:62:3: warning: ‘g_static_mutex_get_mutex_impl’ is deprecated: Use 'GMutex' instead [-Wdeprecated-declarations]
   g_static_mutex_lock(&ffmpeg_mutex);
   ^~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/glib-2.0/gmodule.h:28,
                 from loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:11:
/usr/include/glib-2.0/glib/deprecated/gthread.h:154:9: note: declared here
 GMutex *g_static_mutex_get_mutex_impl (GStaticMutex *mutex);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:67:5: warning: ‘g_static_mutex_get_mutex_impl’ is deprecated: Use 'GMutex' instead [-Wdeprecated-declarations]
     g_static_mutex_unlock(&ffmpeg_mutex);
     ^~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/glib-2.0/gmodule.h:28,
                 from loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:11:
/usr/include/glib-2.0/glib/deprecated/gthread.h:154:9: note: declared here
 GMutex *g_static_mutex_get_mutex_impl (GStaticMutex *mutex);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:72:5: warning: ‘g_static_mutex_get_mutex_impl’ is deprecated: Use 'GMutex' instead [-Wdeprecated-declarations]
     g_static_mutex_unlock(&ffmpeg_mutex);
     ^~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/glib-2.0/gmodule.h:28,
                 from loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:11:
/usr/include/glib-2.0/glib/deprecated/gthread.h:154:9: note: declared here
 GMutex *g_static_mutex_get_mutex_impl (GStaticMutex *mutex);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:80:5: warning: ‘codec’ is deprecated [-Wdeprecated-declarations]
     if (ih->format_context->streams[j]->codec->codec_type
     ^~
In file included from loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:3:
/usr/include/ffmpeg/libavformat/avformat.h:877:21: note: declared here
     AVCodecContext *codec;
                     ^~~~~
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:88:5: warning: ‘g_static_mutex_get_mutex_impl’ is deprecated: Use 'GMutex' instead [-Wdeprecated-declarations]
     g_static_mutex_unlock(&ffmpeg_mutex);
     ^~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/glib-2.0/gmodule.h:28,
                 from loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:11:
/usr/include/glib-2.0/glib/deprecated/gthread.h:154:9: note: declared here
 GMutex *g_static_mutex_get_mutex_impl (GStaticMutex *mutex);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:92:3: warning: ‘codec’ is deprecated [-Wdeprecated-declarations]
   ih->codec_context = ih->format_context->streams[ih->audio_stream]->codec;
   ^~
In file included from loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:3:
/usr/include/ffmpeg/libavformat/avformat.h:877:21: note: declared here
     AVCodecContext *codec;
                     ^~~~~
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:99:5: warning: ‘g_static_mutex_get_mutex_impl’ is deprecated: Use 'GMutex' instead [-Wdeprecated-declarations]
     g_static_mutex_unlock(&ffmpeg_mutex);
     ^~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/glib-2.0/gmodule.h:28,
                 from loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:11:
/usr/include/glib-2.0/glib/deprecated/gthread.h:154:9: note: declared here
 GMutex *g_static_mutex_get_mutex_impl (GStaticMutex *mutex);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:113:5: warning: ‘g_static_mutex_get_mutex_impl’ is deprecated: Use 'GMutex' instead [-Wdeprecated-declarations]
     g_static_mutex_unlock(&ffmpeg_mutex);
     ^~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/glib-2.0/gmodule.h:28,
                 from loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:11:
/usr/include/glib-2.0/glib/deprecated/gthread.h:154:9: note: declared here
 GMutex *g_static_mutex_get_mutex_impl (GStaticMutex *mutex);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:124:5: warning: ‘g_static_mutex_get_mutex_impl’ is deprecated: Use 'GMutex' instead [-Wdeprecated-declarations]
     g_static_mutex_unlock(&ffmpeg_mutex);
     ^~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/glib-2.0/gmodule.h:28,
                 from loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:11:
/usr/include/glib-2.0/glib/deprecated/gthread.h:154:9: note: declared here
 GMutex *g_static_mutex_get_mutex_impl (GStaticMutex *mutex);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:132:3: warning: ‘g_static_mutex_get_mutex_impl’ is deprecated: Use 'GMutex' instead [-Wdeprecated-declarations]
   g_static_mutex_unlock(&ffmpeg_mutex);
   ^~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/glib-2.0/gmodule.h:28,
                 from loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:11:
/usr/include/glib-2.0/glib/deprecated/gthread.h:154:9: note: declared here
 GMutex *g_static_mutex_get_mutex_impl (GStaticMutex *mutex);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:141:3: warning: ‘g_static_mutex_get_mutex_impl’ is deprecated: Use 'GMutex' instead [-Wdeprecated-declarations]
   g_static_mutex_lock(&ffmpeg_mutex);
   ^~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/glib-2.0/gmodule.h:28,
                 from loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:11:
/usr/include/glib-2.0/glib/deprecated/gthread.h:154:9: note: declared here
 GMutex *g_static_mutex_get_mutex_impl (GStaticMutex *mutex);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:143:3: warning: ‘g_static_mutex_get_mutex_impl’ is deprecated: Use 'GMutex' instead [-Wdeprecated-declarations]
   g_static_mutex_unlock(&ffmpeg_mutex);
   ^~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/glib-2.0/gmodule.h:28,
                 from loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:11:
/usr/include/glib-2.0/glib/deprecated/gthread.h:154:9: note: declared here
 GMutex *g_static_mutex_get_mutex_impl (GStaticMutex *mutex);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c: In function ‘decode_packet’:
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:203:3: warning: ‘avcodec_decode_audio4’ is deprecated [-Wdeprecated-declarations]
   ret = avcodec_decode_audio4(ih->codec_context, ih->frame,
   ^~~
In file included from /usr/include/ffmpeg/libavformat/avformat.h:319,
                 from loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:3:
/usr/include/ffmpeg/libavcodec/avcodec.h:4658:5: note: declared here
 int avcodec_decode_audio4(AVCodecContext *avctx, AVFrame *frame,
     ^~~~~~~~~~~~~~~~~~~~~
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c: In function ‘ffmpeg_read_one_packet’:
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:236:7: warning: ‘av_free_packet’ is deprecated [-Wdeprecated-declarations]
       av_free_packet(&ih->packet);
       ^~~~~~~~~~~~~~
In file included from /usr/include/ffmpeg/libavformat/avformat.h:319,
                 from loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:3:
/usr/include/ffmpeg/libavcodec/avcodec.h:4379:6: note: declared here
 void av_free_packet(AVPacket *pkt);
      ^~~~~~~~~~~~~~
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:255:5: warning: ‘av_free_packet’ is deprecated [-Wdeprecated-declarations]
     av_free_packet(&ih->orig_packet);
     ^~~~~~~~~~~~~~
In file included from /usr/include/ffmpeg/libavformat/avformat.h:319,
                 from loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:3:
/usr/include/ffmpeg/libavcodec/avcodec.h:4379:6: note: declared here
 void av_free_packet(AVPacket *pkt);
      ^~~~~~~~~~~~~~
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c: In function ‘ffmpeg_close_file’:
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:362:3: warning: ‘g_static_mutex_get_mutex_impl’ is deprecated: Use 'GMutex' instead [-Wdeprecated-declarations]
   g_static_mutex_lock(&ffmpeg_mutex);
   ^~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/glib-2.0/gmodule.h:28,
                 from loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:11:
/usr/include/glib-2.0/glib/deprecated/gthread.h:154:9: note: declared here
 GMutex *g_static_mutex_get_mutex_impl (GStaticMutex *mutex);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:365:3: warning: ‘g_static_mutex_get_mutex_impl’ is deprecated: Use 'GMutex' instead [-Wdeprecated-declarations]
   g_static_mutex_unlock(&ffmpeg_mutex);
   ^~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/glib-2.0/gmodule.h:28,
                 from loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:11:
/usr/include/glib-2.0/glib/deprecated/gthread.h:154:9: note: declared here
 GMutex *g_static_mutex_get_mutex_impl (GStaticMutex *mutex);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c: In function ‘ffmpeg_init_library’:
loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:370:3: warning: ‘av_register_all’ is deprecated [-Wdeprecated-declarations]
   av_register_all();
   ^~~~~~~~~~~~~~~
In file included from loudness-scanner/scanner/inputaudio/ffmpeg/input_ffmpeg.c:3:
/usr/include/ffmpeg/libavformat/avformat.h:2025:6: note: declared here
 void av_register_all(void);
      ^~~~~~~~~~~~~~~
make[2]: *** [scanner/inputaudio/ffmpeg/CMakeFiles/input_ffmpeg.dir/build.make:63: scanner/inputaudio/ffmpeg/CMakeFiles/input_ffmpeg.dir/input_ffmpeg.c.o] Error 1
[ 75%] Building C object scanner/inputaudio/ffmpeg/CMakeFiles/input_ffmpeg.dir/__/input_helper.c.o
make[2]: Target 'scanner/inputaudio/ffmpeg/CMakeFiles/input_ffmpeg.dir/build' not remade because of errors.
make[1]: *** [CMakeFiles/Makefile2:549: scanner/inputaudio/ffmpeg/CMakeFiles/input_ffmpeg.dir/all] Error 2
Scanning dependencies of target input_gstreamer
[ 77%] Building C object scanner/inputaudio/gstreamer/CMakeFiles/input_gstreamer.dir/input_gstreamer.c.o
loudness-scanner/scanner/inputaudio/gstreamer/input_gstreamer.c: In function ‘bus_call’:
loudness-scanner/scanner/inputaudio/gstreamer/input_gstreamer.c:54:7: warning: ‘g_static_mutex_get_mutex_impl’ is deprecated: Use 'GMutex' instead [-Wdeprecated-declarations]
       g_static_mutex_lock(&gstreamer_mutex);
       ^~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/gstreamer-0.10/gst/gst.h:27,
                 from loudness-scanner/scanner/inputaudio/gstreamer/input_gstreamer.c:3:
/usr/include/glib-2.0/glib/deprecated/gthread.h:154:9: note: declared here
 GMutex *g_static_mutex_get_mutex_impl (GStaticMutex *mutex);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
loudness-scanner/scanner/inputaudio/gstreamer/input_gstreamer.c:59:7: warning: ‘g_static_mutex_get_mutex_impl’ is deprecated: Use 'GMutex' instead [-Wdeprecated-declarations]
       g_static_mutex_unlock(&gstreamer_mutex);
       ^~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/gstreamer-0.10/gst/gst.h:27,
                 from loudness-scanner/scanner/inputaudio/gstreamer/input_gstreamer.c:3:
/usr/include/glib-2.0/glib/deprecated/gthread.h:154:9: note: declared here
 GMutex *g_static_mutex_get_mutex_impl (GStaticMutex *mutex);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
loudness-scanner/scanner/inputaudio/gstreamer/input_gstreamer.c:73:7: warning: ‘g_static_mutex_get_mutex_impl’ is deprecated: Use 'GMutex' instead [-Wdeprecated-declarations]
       g_static_mutex_lock(&gstreamer_mutex);
       ^~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/gstreamer-0.10/gst/gst.h:27,
                 from loudness-scanner/scanner/inputaudio/gstreamer/input_gstreamer.c:3:
/usr/include/glib-2.0/glib/deprecated/gthread.h:154:9: note: declared here
 GMutex *g_static_mutex_get_mutex_impl (GStaticMutex *mutex);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
loudness-scanner/scanner/inputaudio/gstreamer/input_gstreamer.c:78:7: warning: ‘g_static_mutex_get_mutex_impl’ is deprecated: Use 'GMutex' instead [-Wdeprecated-declarations]
       g_static_mutex_unlock(&gstreamer_mutex);
       ^~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/gstreamer-0.10/gst/gst.h:27,
                 from loudness-scanner/scanner/inputaudio/gstreamer/input_gstreamer.c:3:
/usr/include/glib-2.0/glib/deprecated/gthread.h:154:9: note: declared here
 GMutex *g_static_mutex_get_mutex_impl (GStaticMutex *mutex);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
loudness-scanner/scanner/inputaudio/gstreamer/input_gstreamer.c: In function ‘gstreamer_open_file’:
loudness-scanner/scanner/inputaudio/gstreamer/input_gstreamer.c:226:3: warning: ‘g_thread_create’ is deprecated: Use 'g_thread_new' instead [-Wdeprecated-declarations]
   ih->gstreamer_loop = g_thread_create((GThreadFunc) gstreamer_loop, ih, TRUE, NULL);
   ^~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/gstreamer-0.10/gst/gst.h:27,
                 from loudness-scanner/scanner/inputaudio/gstreamer/input_gstreamer.c:3:
/usr/include/glib-2.0/glib/deprecated/gthread.h:104:10: note: declared here
 GThread *g_thread_create       (GThreadFunc       func,
          ^~~~~~~~~~~~~~~
[ 80%] Linking C shared module ../../../libinput_gstreamer.so
[ 80%] Built target input_gstreamer
Scanning dependencies of target loudness-drop-gtk
[ 83%] Building C object scanner/scanner-drop-gtk/CMakeFiles/loudness-drop-gtk.dir/scanner-drop.c.o
In file included from loudness-scanner/scanner/scanner-drop-gtk/scanner-drop.c:6:
/usr/include/librsvg-2.0/librsvg/rsvg-cairo.h:27:2: warning: #warning "Including <librsvg/rsvg-cairo.h> directly is deprecated." [-Wcpp]
 #warning "Including <librsvg/rsvg-cairo.h> directly is deprecated."
  ^~~~~~~
loudness-scanner/scanner/scanner-drop-gtk/scanner-drop.c: In function ‘do_work’:
loudness-scanner/scanner/scanner-drop-gtk/scanner-drop.c:355:5: warning: ‘g_static_mutex_get_mutex_impl’ is deprecated: Use 'GMutex' instead [-Wdeprecated-declarations]
     g_static_mutex_lock(&thread_mutex);
     ^~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from loudness-scanner/scanner/scanner-drop-gtk/scanner-drop.c:3:
/usr/include/glib-2.0/glib/deprecated/gthread.h:154:9: note: declared here
 GMutex *g_static_mutex_get_mutex_impl (GStaticMutex *mutex);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
loudness-scanner/scanner/scanner-drop-gtk/scanner-drop.c:357:5: warning: ‘g_static_mutex_get_mutex_impl’ is deprecated: Use 'GMutex' instead [-Wdeprecated-declarations]
     g_static_mutex_unlock(&thread_mutex);
     ^~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from loudness-scanner/scanner/scanner-drop-gtk/scanner-drop.c:3:
/usr/include/glib-2.0/glib/deprecated/gthread.h:154:9: note: declared here
 GMutex *g_static_mutex_get_mutex_impl (GStaticMutex *mutex);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
loudness-scanner/scanner/scanner-drop-gtk/scanner-drop.c: In function ‘handle_data_received’:
loudness-scanner/scanner/scanner-drop-gtk/scanner-drop.c:375:5: warning: ‘g_static_mutex_get_mutex_impl’ is deprecated: Use 'GMutex' instead [-Wdeprecated-declarations]
     g_static_mutex_lock(&thread_mutex);
     ^~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from loudness-scanner/scanner/scanner-drop-gtk/scanner-drop.c:3:
/usr/include/glib-2.0/glib/deprecated/gthread.h:154:9: note: declared here
 GMutex *g_static_mutex_get_mutex_impl (GStaticMutex *mutex);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
loudness-scanner/scanner/scanner-drop-gtk/scanner-drop.c:377:9: warning: ‘g_static_mutex_get_mutex_impl’ is deprecated: Use 'GMutex' instead [-Wdeprecated-declarations]
         g_static_mutex_unlock(&thread_mutex);
         ^~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from loudness-scanner/scanner/scanner-drop-gtk/scanner-drop.c:3:
/usr/include/glib-2.0/glib/deprecated/gthread.h:154:9: note: declared here
 GMutex *g_static_mutex_get_mutex_impl (GStaticMutex *mutex);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
loudness-scanner/scanner/scanner-drop-gtk/scanner-drop.c:381:5: warning: ‘g_static_mutex_get_mutex_impl’ is deprecated: Use 'GMutex' instead [-Wdeprecated-declarations]
     g_static_mutex_unlock(&thread_mutex);
     ^~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from loudness-scanner/scanner/scanner-drop-gtk/scanner-drop.c:3:
/usr/include/glib-2.0/glib/deprecated/gthread.h:154:9: note: declared here
 GMutex *g_static_mutex_get_mutex_impl (GStaticMutex *mutex);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
loudness-scanner/scanner/scanner-drop-gtk/scanner-drop.c:395:5: warning: ‘g_thread_create’ is deprecated: Use 'g_thread_new' instead [-Wdeprecated-declarations]
     worker_thread = g_thread_create((GThreadFunc) do_work, sl, FALSE, NULL);
     ^~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from loudness-scanner/scanner/scanner-drop-gtk/scanner-drop.c:3:
/usr/include/glib-2.0/glib/deprecated/gthread.h:104:10: note: declared here
 GThread *g_thread_create       (GThreadFunc       func,
          ^~~~~~~~~~~~~~~
loudness-scanner/scanner/scanner-drop-gtk/scanner-drop.c: In function ‘main’:
loudness-scanner/scanner/scanner-drop-gtk/scanner-drop.c:564:5: warning: ‘g_thread_init’ is deprecated [-Wdeprecated-declarations]
     g_thread_init(NULL);
     ^~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from loudness-scanner/scanner/scanner-drop-gtk/scanner-drop.c:3:
/usr/include/glib-2.0/glib/deprecated/gthread.h:265:10: note: declared here
 void     g_thread_init                   (gpointer vtable);
          ^~~~~~~~~~~~~
loudness-scanner/scanner/scanner-drop-gtk/scanner-drop.c:614:5: warning: ‘g_thread_create’ is deprecated: Use 'g_thread_new' instead [-Wdeprecated-declarations]
     bar_thread = g_thread_create((GThreadFunc) update_bar, window, FALSE, NULL);
     ^~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:108,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from loudness-scanner/scanner/scanner-drop-gtk/scanner-drop.c:3:
/usr/include/glib-2.0/glib/deprecated/gthread.h:104:10: note: declared here
 GThread *g_thread_create       (GThreadFunc       func,
          ^~~~~~~~~~~~~~~
[ 86%] Linking CXX executable ../../loudness-drop-gtk
[ 86%] Built target loudness-drop-gtk
Scanning dependencies of target loudness-drop-qt
[ 88%] Building CXX object scanner/scanner-drop-qt/CMakeFiles/loudness-drop-qt.dir/scanner-drop-qt.cpp.o
loudness-scanner/scanner/scanner-drop-qt/scanner-drop-qt.cpp: In function ‘int main(int, char**)’:
loudness-scanner/scanner/scanner-drop-qt/scanner-drop-qt.cpp:500:23: warning: ‘void g_thread_init(gpointer)’ is deprecated [-Wdeprecated-declarations]
     g_thread_init(NULL);
                       ^
In file included from /usr/include/glib-2.0/glib.h:108,
                 from loudness-scanner/scanner/filetree/filetree.h:6,
                 from loudness-scanner/scanner/scanner-drop-qt/scanner-drop-qt.h:19,
                 from loudness-scanner/scanner/scanner-drop-qt/scanner-drop-qt.cpp:3:
/usr/include/glib-2.0/glib/deprecated/gthread.h:265:10: note: declared here
 void     g_thread_init                   (gpointer vtable);
          ^~~~~~~~~~~~~
[ 91%] Building CXX object scanner/scanner-drop-qt/CMakeFiles/loudness-drop-qt.dir/moc_scanner-drop-qt4.cpp.o
[ 94%] Linking CXX executable ../../loudness-drop-qt
[ 97%] Built target loudness-drop-qt
make[1]: Target 'all' not remade because of errors.
make: *** [Makefile:130: all] Error 2
make: Target 'default_target' not remade because of errors.

```

**Comment by zuzzurro** [comment]
OK, so I went ahead and tried fixing the first issue. According to the info I found on the web FF_INPUT_BUFFER_PADDING_SIZE is gone and has been replaced by AV_INPUT_BUFFER_PADDING_SIZE.
This change alone let me compile everything (with warnings).

**Comment by zuzzurro** [comment]
Created a pull request with a couple of changes that fix the compilation for me: #30

**Comment by jiixyj** [comment]
I've merged the PR just now. Thanks again!
